### PR TITLE
chore(zod): add zod regex to project routes && update generic zod

### PR DESCRIPTION
**Pull request by janeyuaws**
Issue #, if available: GALI-2324

Description of changes:
* add zod to proj route
* update ZodString interface
* update unit test and integ test
* update regex

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-reference/integration-tests/tests/isolated/projects/create.test.ts [hunk 1]
```diff
@@ -121,7 +126,7 @@ describe('Create Project negative tests', () => {
   describe('with a cost center', () => {
     describe('that does not exist', () => {
       beforeEach(async () => {
-        createRequest.costCenterId = 'cc-invalid-cost-center';
```
Comment (SanketD92) [inline solutions/swb-reference/integration-tests/tests/isolated/projects/create.test.ts:124]: Could you add another test case that covers the invalid CC ID format?

File: workbench-core/accounts/src/models/projects/assignUserToProjectRequest.ts [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
  */
 
-import { z } from 'zod';
+import { resourceTypeToKey, z } from '@aws/workbench-core-base';
 
 // eslint-disable-next-line @rushstack/typedef-var
 export const AssignUserToProjectRequestParser = z
   .object({
-    projectId: z.string(),
+    projectId: z.string().swbId(resourceTypeToKey.project.toLowerCase()).required(),
```
Comment (SanketD92) [inline workbench-core/accounts/src/models/projects/assignUserToProjectRequest.ts:11]: If my [PR](https://github.com/aws-solutions/solution-spark-on-aws/pull/1045) merges first, you can replace this with `z.string.projId().required()`

File: workbench-core/accounts/src/models/projects/createProjectRequest.ts [hunk 3]
```diff
@@ -2,14 +2,14 @@
...
 
 // eslint-disable-next-line @rushstack/typedef-var
 export const CreateProjectRequestParser = z
   .object({
-    name: z.string(),
-    description: z.string(),
-    costCenterId: z.string()
+    name: z.string().swbName().required(),
+    description: z.string().swbDescription().required(),
+    costCenterId: z.string().swbId(resourceTypeToKey.costCenter.toLowerCase()).required()
```
Comment (SanketD92) [inline workbench-core/accounts/src/models/projects/createProjectRequest.ts:12]: Create a new ZodString interface function for `ccId` in `workbench-core/base/src/utilities/validatorHelper.ts`

File: workbench-core/accounts/src/models/projects/deleteProjectRequest.ts [hunk 4]
```diff
@@ -4,13 +4,13 @@
...
 import { AuthenticatedUserParser } from '@aws/workbench-core-authorization';
-import { z } from 'zod';
+import { resourceTypeToKey, z } from '@aws/workbench-core-base';
 
 // eslint-disable-next-line @rushstack/typedef-var
 export const DeleteProjectRequestParser = z
   .object({
     authenticatedUser: AuthenticatedUserParser,
-    projectId: z.string()
+    projectId: z.string().swbId(resourceTypeToKey.project.toLowerCase()).required()
```
Comment (SanketD92) [inline workbench-core/accounts/src/models/projects/deleteProjectRequest.ts:13]: same comment as above

File: workbench-core/accounts/src/models/projects/getProjectRequest.ts [hunk 5]
```diff
@@ -2,12 +2,12 @@
...
  *  SPDX-License-Identifier: Apache-2.0
  */
-import { z } from 'zod';
+import { resourceTypeToKey, z } from '@aws/workbench-core-base';
 
 // eslint-disable-next-line @rushstack/typedef-var
 export const GetProjectRequestParser = z
   .object({
-    projectId: z.string()
+    projectId: z.string().swbId(resourceTypeToKey.project.toLowerCase()).required()
```
Comment (SanketD92) [inline workbench-core/accounts/src/models/projects/getProjectRequest.ts:10]: same comment as above

**Review by insignias — APPROVED** [review]

**Pull request merged**